wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -73,6 +75,26 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
   override def listAddresses(): Future[Vector[AddressDb]] =
     addressDAO.findAll()
 
+  /** Enumerates the public keys in this wallet */
+  private[wallet] def listPubkeys(): Future[Vector[ECPublicKey]] =
+    addressDAO.findAllPubkeys().map(_.toVector)
+
+  // todo: insert TXIDs? need to track which txids we should
+  // ask for, somehow
```

Comment:
So basically we need a table for transactions we have created and signed? 

An side note here:

We should _always_ treat a signed transaction as broadcasted, I don't think we should model the state transitition as 

1. Unsigned tx
2. Signed but not broadcasted
3. Signed and broadcasted
4. Confirmed

Reply:
Yeah so this is more for future reference, once that gets added. Thinking we do the stupid thing for now, which is just scan for money on pubkeys we own. Otherwise I agree with your comment